gallery/tutorials/ctf.py

```diff
@@ -0,0 +1,414 @@
+    B=0,
+)
+# Evaluate Filter, returning a numpy array.
+bad_ctf_fn = bad_est_ctf_filter.evaluate_grid(RESOLUTION)
+
+c = RESOLUTION // 2 + 1
+plt.plot(rctf_fn[c, c:], label="Model CTF")  # radial_ctf_filter
+plt.plot(bad_ctf_fn[c, c:], label="Incorrect CTF Estimate")
+plt.legend()
+plt.show()
+
+# %%
+# Compare the idealized CTF phase flipping correction
+# with phase flipping an incorrect CTF.
+
+idealized_flipped_fn = rctf_fn * np.sign(rctf_fn)
+incorrect_flipped_fn = rctf_fn * np.sign(bad_ctf_fn)
+fig, (ax1, ax2) = plt.subplots(1, 2, sharex=True, sharey=True)
+fig.suptitle("Application of Phase Flipping")
+ax1.plot(idealized_flipped_fn[c, c:])
+ax1.set_title("Idealized")
+ax2.plot(incorrect_flipped_fn[c, c:])
+ax2.set_title("Incorrect")
+plt.show()
+
+# %%
+# ASPIRE CtfEstimator
+# ^^^^^^^^^^^^^^^^^^^
+# Here we will use ASPIRE's ``CtfEstimator`` on a CTF corrupted image
+# so we can compare with the modeled corruption parameters.
+
+from aspire.ctf import estimate_ctf
+
+# Using our radial_ctf_filter from earlier, corrupt an image.
+test_img = Image(img).filter(radial_ctf_filter)
+plt.imshow(test_img.asnumpy()[0])
+plt.colorbar()
+plt.show()
+
+# Create the image file in a tmp dir
+with TemporaryDirectory() as d:
+    test_img.save(os.path.join(d, "test_img.mrc"))
+
+    radial_ctf_est = estimate_ctf(
+        data_folder=d,
+        pixel_size=radial_ctf_filter.pixel_size,
+        cs=radial_ctf_filter.Cs,
+        amplitude_contrast=radial_ctf_filter.alpha,
+        voltage=radial_ctf_filter.voltage,
+        psd_size=512,
+        num_tapers=2,
+        dtype=np.float64,
+    )
+
+# We'll use these estimates next.
+print(radial_ctf_est)
+
+# %%
+# Using ASPIRE's CTF Estimate
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Previously we computed some radial CTF estimates in ``radial_ctf_est``.
+# Here we will view the application of these estimates in phase flipping
+# our example image.
+
+# Get the relevant estimate.
+est = radial_ctf_est["test_img.mrc"]
+
+# Take an average defocus for radial case.
+defocus = (est["defocus_u"] + est["defocus_v"]) / 2.0
+print(f"defocus = {defocus}")
+
+# Create a filter and evaluate.
+est_ctf = RadialCTFFilter(
+    pixel_size=est["pixel_size"],
+    voltage=est["voltage"],
+    defocus=defocus,  # Modeled CTF was 10000
+    Cs=est["cs"],
+    alpha=est["amplitude_contrast"],
+    B=0,
+)
+est_ctf_fn = est_ctf.evaluate_grid(RESOLUTION)
+
+# Compare the model CTF with the estimated CTF.
+c = RESOLUTION // 2 + 1
+plt.plot(rctf_fn[c, c:], label="Model CTF")
+plt.plot(est_ctf_fn[c, c:], label="Estimated CTF")
+plt.legend()
+plt.show()
+
+# %%
+# Compare the idealized CTF phase flipping correction
+# with phase flipping the estimated CTF.
+
+estimated_flipped_fn = rctf_fn * np.sign(est_ctf_fn)
+plt.title("Application of estimated CTF Phase Flipping")
+plt.plot(estimated_flipped_fn[c, c:])
+plt.show()
+
+# %%
+# .. note::
+#     At the time of writing, asymmetric CTF estimation is under going development and validation.
```

Comment:
If changed above, change "asymmetric" here too.

Reply:
changed to elliptical